src/main/kotlin/uk/gov/dluhc/printapi/service/PrintResponseProcessingService.kt

```diff
@@ -72,8 +78,22 @@ class PrintResponseProcessingService(
     }
 
     fun processPrintResponse(printResponse: ProcessPrintResponseMessage) {
-        val newStatus = statusMapper.toStatusEntityEnum(printResponse.statusStep, printResponse.status)
-        val printDetails = printDetailsRepository.getByRequestId(printResponse.requestId)
+        val newStatus: Status
+        val printDetails: PrintDetails
+
+        try {
+            newStatus = statusMapper.toStatusEntityEnum(printResponse.statusStep, printResponse.status)
+            printDetails = printDetailsRepository.getByRequestId(printResponse.requestId)
+        } catch (ex: Exception) {
```

Comment:
Not sure about this exception handling ... 🤔 
I'm not a fan of catching `Exception` if we can help it? It feels too broad? Is there something more specific we can catch here?
Secondly I'm not sure what we need to do in the case of an exception here. I guess its all about the context of this method and how and when it is called.

As I understand it, its called by the listener for a single print response message by its `requestId`. Its about processing 1 and only 1 print response.

If this were to fail, how might it fail, and what do we want to do?

As you have it at the moment, if the mapper fails (likely scenario ??), or it fails to find the record in the database, then we'll log an `error` and then return. Which means the SQS message will successfully ACK and will be considered complete. I guess this is right?? Or, so we want to to go straight to the DLQ do it can be investigated? (as it stands at the moment it wont DLQ so cant be investigated / re-played)

If the exception is something else (not sure what ?), then we re-throw. As per my comments about this will cause the message to be re-queued until and retried up to 5 times, by which time it hits the DLQ.
Maybe thats right? I guess perhaps for transient errors such as database gone down etc?

What do you think? 🤔 
Any thoughts @Neil-Massey ?

Reply:
From talking with @stalwartsolutions yesterday it seems that logs are not monitored but DLQ queues are.  So if we want to be aware of issues we need messages to end in the DLQ.  Messages that end in the DLQ can be replayed at a later stage - perhaps after a code or infra fix.  

Perhaps at this stage allowing all errors to be retried so either they succeed on the retry or end in the DLQ is the best approach (instead of logging the error, handling the exception and removing the message from the queue) - but as you say this is only if there is required processing still required.

Reply:
@NathanRussellValtech , @Neil-Massey , thanks for your comments. 
The only cases we log and don't throw an exception, hence the message not ending up in the DLQ, are:

- There is no printRequest for the given `requestId` in the DB
- The StatusStep is `UNDELIVERED` but step is `SUCCESS`, which is not a valid defined scenario.

In both cases, no code or infra change will fix the issues, and I don't think there is a logical way to process that `printResponse`. 
So, maybe we log for those cases and carry on as suggested in the ticket? 
Does that make sense?